lib/lua/checkStalledJobs.lua

```diff
@@ -31,20 +35,24 @@ if next(stalling) ~= nil then
       stalled[#stalled + 1] = jobId
     end
   end
+  
+  -- lpush instead of rpush so that jobs which cause uncaught exceptions don't
+  -- hog the job consumers and starve the whole system. not a great situation
+  -- to be in, but this is fairer.
+  local pushed = 0
   -- don't lpush zero jobs (the redis command will fail)
-  if #stalled > 0 then
-    -- lpush instead of rpush so that jobs which cause uncaught exceptions don't
-    -- hog the job consumers and starve the whole system. not a great situation
-    -- to be in, but this is fairer.
-    redis.call("lpush", KEYS[3], unpack(stalled))
+  while (pushed < #stalled) do
```

Comment:
The `#` length operator is re-computed on each reference - we should push this (and the later `#actives` condition) into a `local` to avoid this penalty.